docs/frontend.md

```diff
@@ -1,5 +1,44 @@
 # Front-end Architecture
 
+While the Login.gov application is architected as a Ruby on Rails project and most of its pages are
+fully rendered by the backend server, the front end is set up to use modern best-practices to suit
+the needs of interactivity and consistency in the user experience, and to ensure a maintainable and
+convenient developer experience.
+
+As a high-level overview, the front end consists of:
+
+- [Propshaft](https://github.com/rails/propshaft) is used as an asset pipeline library for Ruby on
+  Rails, in combination with [cssbundling-rails](https://github.com/rails/cssbundling-rails) and
+  [jsbundling-rails](https://github.com/rails/jsbundling-rails) for compiling stylesheets and
+  scripts.
+- JavaScript is written as [TypeScript](https://www.typescriptlang.org/), bundled using [Webpack](https://webpack.js.org/)
+  with syntax transformation applied by [Babel](https://babeljs.io/).
+   - For highly-interactive pages, we use [React](https://react.dev/)
+   - For all other JavaScript interaction, we use native [web components](https://developer.mozilla.org/en-US/docs/Web/API/Web_components)
+     (custom elements)
+- Stylesheets are written as [Sass](https://sass-lang.com/), and builds upon the [Login.gov Design System](https://github.com/18F/identity-design-system),
+  which in turn builds upon the [U.S. Web Design System (USWDS)](https://designsystem.digital.gov/).
+- HTML reuse is facilitated by the [ViewComponent](https://viewcomponent.org/) gem, enhanced with
+  support for per-component JavaScript and stylesheet "sidecar" assets.
```

Comment:
I am not familiar with `stylesheet "sidecar" assets`.  Could you please include a brief explanation of the term or a link to a definition?  Thanks.

Reply:
I waffled on whether to include this "sidecar" term. It's a [term used in ViewComponent](https://viewcomponent.org/api.html#sidecar_filesextensions) to refer to other assets that are expected to be loaded alongside a component (stylesheets, scripts). 

It's described in more detail in the [components directory README](https://github.com/18F/identity-idp/blob/main/app/components/README.md), maybe I can update the text here to link to that article.

Reply:
Sure, linking to another part of the docs that describes the term would work.

Reply:
I removed this second part of the sentence in 12da45f. The changes in 11ba76a add a link to the components README for more information. In follow-up pull requests I'm going to flesh out the subdirectory READMEs a bit more, so I might build upon what's already in the components README to describe the assets behavior a bit more fully.